flavors/publisher.go

```diff
@@ -0,0 +1,94 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package flavors
+
+import (
+	"context"
+	"time"
+
+	"github.com/elastic/beats/v7/libbeat/beat"
+	"github.com/elastic/elastic-agent-libs/logp"
+)
+
+type client interface {
+	PublishAll([]beat.Event)
+}
+
+type Publisher struct {
+	log      *logp.Logger
+	interval time.Duration
+	treshold int
+	client   client
+}
+
+func NewPublisher(log *logp.Logger, interval time.Duration, treshold int, client client) *Publisher {
+	return &Publisher{
+		log:      log,
+		interval: interval,
+		treshold: treshold,
+		client:   client,
+	}
+}
+
+func (p *Publisher) HandleEvents(ctx context.Context, ch chan []beat.Event) {
+	var eventsToSend []beat.Event
+	flushTicker := time.NewTicker(p.interval)
+	for {
+		select {
+		case <-ctx.Done():
+			p.log.Warn("Publisher context is done")
```

Comment:
```suggestion
			p.log.Warnf("Publisher context is done: %v", ctx.Err())
```

slightly more informative